### DISCUSSION TITLE
Seismic modelling from EBSD data and a single orientation

### DISCUSSION THREAD

**Discussion by Liujianhua00** [mtex-toolbox/mtex#1780] [Ask Anything]
Hi, everyone,
I have a question about the seismic modelling using MTEX, I calculate the seismic properties using two types of data. One comes from the EBSD data of natural sample, and the other one comes from a single orientation I defined in MTEX. I calculate ODFs from both of them than use the same stiffnessTensor to model the seismic properties. I also plot the pole figures and they have the same CPO (crystal preferred orientation) type. However.  I find that the symmetry of seismic anisotropy between them is different. 

  If the two data have same CPO type in pole figure, they should have same symmetry of seismic anisotropy. But the symmetry of seismic anisotropy from a single orientation more likely the anisotropy of amphibole single crystal. So will this difference comes from the transformation of reference frame or some other reasons? Thanks a lot in advance!

![1](https://user-images.githubusercontent.com/114222474/227566722-f17156f7-fc9c-46cf-844e-5f599afd6dee.png)
![2](https://user-images.githubusercontent.com/114222474/227566796-0fe3f1e8-0310-4337-8193-83d58c501b92.png)
![3](https://user-images.githubusercontent.com/114222474/227566821-5e20261f-7b89-479a-8f1c-13f33a3998e6.png)
![4](https://user-images.githubusercontent.com/114222474/227566851-1e34f50d-c3a6-4a78-b724-fbf4e60a8271.png)

The part of code I used to calculated seismic property from a single orientation. 

`ori = orientation.byEuler(0*degree,90*degree, 0*degree, cs_Tensor)
odf = calcDensity(ori,'halfwidth',10*degree);
[CVoigt, CReuss, CHill] = calcTensor(odf,C);
close all
plotSeismicVelocities(CHill)
saveFigure ('seismic velocities aggregate.png')`

**Comment by lfgmorales** [comment]
Hi

The calculations based on EBSD data seem to be correct, but I think your approach for modelling your “single orientation” is not really correct, because if you want to plot the single crystal seismic properties, you should do something like

% the crystal reference system
cs_Tensor_opx = crystalSymmetry('mmm',[ 18.2457  8.7984  5.1959],...
  [  90.0000  90.0000  90.0000]*degree,'x||a','z||c',...
  'mineral','Enstatite');

% the density
rho_opx = 3.3060;

% the tensor coefficients
Cij =....
  [[  236.90   79.60   63.20    0.00    0.00    0.00];...
  [    79.60  180.50   56.80    0.00    0.00    0.00];...
  [    63.20   56.80  230.40    0.00    0.00    0.00];...
  [     0.00    0.00    0.00   84.30    0.00    0.00];...
  [     0.00    0.00    0.00    0.00   79.40    0.00];...
  [     0.00    0.00    0.00    0.00    0.00   80.10]];

% define the tensor
C_opx = stiffnessTensor(Cij,cs_Tensor_opx,'density',rho_opx);
And then

plotSeismicVelocities(C_Opx)
Without the need of calculating an odf.

The fact that everything looks asymmetrical in your “single crystal” either comes from the fact that the components of your elastic constant do not fulfil the symmetry (unlikely, as MTEX throws an error or a warning), or more likely because your single crystal ODF is not correct, you use a halfwidth of 10 degrees, which probably leads to noise in the ODF, and result in this more complex seismic wave distribution (the values seem correct at least). 


cheers


Luiz

**Reply by Liujianhua00** [reply]
Thanks very much for your reply. In fact, I also tried to directly use the single orientation (without the calculating an ODF) to calculate the seismic anisotropy, using the following code. 

 `[CVoigt, CReuss, CHill] = calcTensor(ori,C_amp);
plotSeismicVelocities(CHill)
saveFigure ('seismic velocities aggregate.png')`

The result is the same with the one using calculated ODF. Moreover, it also should be the seismic anisotropy of amphibole single crystal.

![5](https://user-images.githubusercontent.com/114222474/227697253-9e237e88-329e-49c0-b6b5-1b3f293a47cc.jpg)

So I think the halfwidth I used may not be reason to the asymmetry of seismic anisotropy.

To find out reason, this time I defined 200 orientations by Euler angle and I calculated the ODF and the seismic property from this data. This is the code and orientations I used, and also the final results. But the asymmetry of seismic anisotropy still occurred.

`%%
cs_Tensor_amp= crystalSymmetry('12/m1', [9.9 18 5.3], [90,105.2,90]*degree, 'X||a*', 'Y||b*', 'Z||c', ...
'mineral', 'Hornblende', 'color', [0.85 0.65 0.13]);
fname = fullfile(mtexDataPath,'orientation','Euler');
ori = orientation.load(fname,'columnNames',{'phi1','Phi','phi2'}, cs_Tensor_amp)
%%
 Cij_amp =.... 
 [[  122.80   51.98   45.90    0.00   -0.70    0.00];...
 [    51.98  189.30   62.30    0.00   -7.00    0.00];...
 [    45.90   62.30  222.90    0.00  -30.00    0.00];...
 [     0.00    0.00    0.00   71.50    0.00    5.40];...
 [    -0.70   -7.00  -30.00    0.00   46.80    0.00];...
 [     0.00    0.00    0.00    5.40    0.00   46.2]];

% Define density (g/cm3)
rho_amp = 3.293;
% define the tensor
C_amp = stiffnessTensor(Cij_amp,cs_Tensor_amp,'density',rho_amp);
%%
odf = calcDensity(ori,'halfwidth',10*degree);
%%
PFs_amp = [ ...
    Miller(1,0,0,cs_Tensor_amp,'hkl'), ...
    Miller(0,1,0,cs_Tensor_amp,'hkl'), ...
    Miller(0,0,1,cs_Tensor_amp,'uvw'), ...
     Miller(1,1,0,cs_Tensor_amp,'hkl'), ...
          ];
figure
plotPDF(ori,PFs_amp,'lower','resolution',5*degree,'contourf','colorrange',...
    'equal','minmax','silent')
%%
[CVoigt, CReuss, CHill] = calcTensor(odf,C_amp);
close all
plotSeismicVelocities(CHill)
saveFigure ('seismic velocities aggregate.png')`
![6](https://user-images.githubusercontent.com/114222474/227697689-7c6ea6b9-166b-49c2-a55f-e0ad262aeb15.jpg)
![7](https://user-images.githubusercontent.com/114222474/227697690-0e894877-cb00-4b46-951b-24e31db9f46f.jpg)

orientations
[Euler.xlsx](https://github.com/mtex-toolbox/mtex/files/11068056/Euler.xlsx)

**Comment by lfgmorales** [comment]
Hi

Your crystal symmetry is wrongly define, you have to have 'x||a*’,'z||c’, if you have y||b* as you wrote, that makes it a non euclidian reference frame, and because of that I am not entirely sure what you are calculating (surely not the amphibole as it throws an error)

I also notice some minor differences in the components of your elastic constant, have a look on the original paper, you can use this one below

%% Define elastic stiffness tensor of hornblende 
% Brown and Abramson, Phys. Earth Planet. Int.(2016) X||a* Y||b Z||c
 M_amph =.... 
 [[  122.80   51.80   45.90    0.00   -0.70    0.00];...
 [    51.80  189.30   62.30    0.00   -7.00    0.00];...
 [    45.90   62.30  222.90    0.00  -30.00    0.00];...
 [     0.00    0.00    0.00   71.50    0.00    5.40];...
 [    -0.70   -7.00  -30.00    0.00   46.80    0.00];...
 [     0.00    0.00    0.00    5.40    0.00   46.20]];
rho_amph= 3.293;
cs_Tensor_amph = crystalSymmetry('12/m1',[ 9.83 18.08 5.32],...
 [ 90, 104.9, 90 ]*degree,'x||a*','y||b','z||c',...
 'mineral','Hornblende-Tshermakite');
C_amph = stiffnessTensor(M_amph,cs_Tensor_amph,'density',rho_amph);
plotSeismicVelocities (C_amph)


About the halfwidth 10 degrees, this is incorrect for a single orientation, have a look in the odf section

https://mtex-toolbox.github.io/ODFTheory.html <https://mtex-toolbox.github.io/ODFTheory.html>

Cheers


Luiz

**Reply by Liujianhua00** [reply]
Until now, I have tested several crystal symmetry of hornblend. Unfortunately, these crystal symmetry make no diffference in the symmetry  of seismic anisotropy. I tried another way to find out the reason.  I pick out an amphibole grain from my EBSD data and export as a new file, and calculated the seismic property of this file. The result is asymmetrical. In fact, the single crystal seismic property of amphibole is also asymmetrical. So I think the symmetry of seismic anisotropy in natural samples may result from statistical calculation of hundreds of thousands of orientations.

`cs_Tensor_amp= crystalSymmetry('12/m1', [9.8 18 5.3], [90,105.05,90]*degree, 'X||a*', 'Y||b*', 'Z||c',...
     'mineral', 'Hornblende', 'color', [0.56 0.74 0.56]);
 cs_Tensor_amp= crystalSymmetry('12/m1', [9.9 18 5.3], [90,105.2,90]*degree, 'X||a*', 'Y||b*', 'Z||c', ...
'mineral', 'Hornblende', 'color', [0.85 0.65 0.13]);
 cs_Tensor_amp = crystalSymmetry('2/m',[9.86,18.07,5.33],[90,105.48,90]*degree,'x||a*','z||c','mineral','hornblende');
 cs_Tensor_amp = crystalSymmetry('12/m1', [9.9 18 5.4], [90,104.81,90]*degree, 'X||a*', 'Y||b', 'Z||c', 'mineral', 'Amphibole', 'color', [0.53 0.81 0.98]);
cs_Tensor_amp  = crystalSymmetry('12/m1',[ 9.83 18.08 5.32], [ 90, 104.9, 90 ]*degree,'x||a*','y||b','z||c', 'mineral','Hornblende-Tshermakite');`

**Comment by lfgmorales** [comment]
Hi,

Amphibole is monoclinic, and therefore you get a monoclinic seismic wave velocity distribution for a single crystal. You should use the symmetry of the elastic constant as given on the experimental measurements, if you don’t do that your results will be incorrwct

Luiz


Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 25 Mar 2023, at 12:05, Liujianhua00 ***@***.***> wrote:

﻿

Until now, I have tested several crystal symmetry of hornblend. Unfortunately, these crystal symmetry make no diffference in the symmetry of seismic anisotropy. I tried another way to find out the reason. I pick out an amphibole grain from my EBSD data and export as a new file, and calculated the seismic property of this file. The result is asymmetrical. In fact, the single crystal seismic property of amphibole is also asymmetrical. So I think the symmetry of seismic anisotropy in natural samples may result from statistical calculation of hundreds of thousands of orientations.

cs_Tensor_amp= crystalSymmetry('12/m1', [9.8 18 5.3], [90,105.05,90]*degree, 'X||a*', 'Y||b*', 'Z||c',... 'mineral', 'Hornblende', 'color', [0.56 0.74 0.56]); cs_Tensor_amp= crystalSymmetry('12/m1', [9.9 18 5.3], [90,105.2,90]*degree, 'X||a*', 'Y||b*', 'Z||c', ... 'mineral', 'Hornblende', 'color', [0.85 0.65 0.13]); cs_Tensor_amp = crystalSymmetry('2/m',[9.86,18.07,5.33],[90,105.48,90]*degree,'x||a*','z||c','mineral','hornblende'); cs_Tensor_amp = crystalSymmetry('12/m1', [9.9 18 5.4], [90,104.81,90]*degree, 'X||a*', 'Y||b', 'Z||c', 'mineral', 'Amphibole', 'color', [0.53 0.81 0.98]); cs_Tensor_amp = crystalSymmetry('12/m1',[ 9.83 18.08 5.32], [ 90, 104.9, 90 ]*degree,'x||a*','y||b','z||c', 'mineral','Hornblende-Tshermakite');

—
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1780#discussioncomment-5425981>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN56AY46LORYGJDVZLDW53GNXANCNFSM6AAAAAAWGW7YQA>.
You are receiving this because you commented.Message ID: ***@***.***>

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1780#discussioncomment-5425981", "url": "https://github.com/mtex-toolbox/mtex/discussions/1780#discussioncomment-5425981", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by Liujianhua00** [reply]
Yes, indeed. I used the elastic stiffness tensor of hornblende you provide to calculate the amphibole grain I picked out. I upload the pole figure and seismic property. As a single grain, the orientations in it should the same. It will lead to seismic property of a single crystal, which is asymmetrical. However, in natural samples, it's a statistical calculation including hundreds of thousands of grains. Thus, the asymmetry of single grains may be covered and present a symmetrical seismic property of whole sample. 

![11](https://user-images.githubusercontent.com/114222474/227715476-9429f7ac-dce2-4f8f-bce9-a4c3c3457ece.png)
![10](https://user-images.githubusercontent.com/114222474/227715479-5dd3e1d3-5213-4a3e-805a-ebb5b5f26f0f.png)
